examples/chip-tool/commands/interactive/InteractiveCommands.cpp

```diff
@@ -158,11 +171,26 @@ struct InteractiveServerResult
     void MaybeAddResult(const char * result)
     {
         auto lock = ScopedLock(mMutex);
-        VerifyOrReturn(mEnabled);
+        // If we are not waiting for result it could be that a report came in *before* the command to wait for
+        // it has happened. The result is stored until the next command. If the next command is an async
+        // command the pending result would be promote to a valid result so it could be dispatched properly.
+        if (!mEnabled)
```

Comment:
So just in terms of ordering.  Say I have YAML that does:

1. Send a command.
2. Wait for a report on a subscription caused by the changes due to that command.

What happens if the report comes before the command response?

Reply:
That is a good question and I don't have a good response handy.
The way it works today is that the report will likely be added to the command response itself.

In this case, we could probably update the python implementation to "consume" the report before feeding the response processor and save it until the next command. If the next command is a wait for report, then we could probably avoid to do a network round trip and give it directly as a response.

It may requires a few more changes to the adapter implementation, especially since the `encoder` and the `decoder` does not communicate at all as of today. Is that required at the moment ?

Reply:
I think it would be good to track as a followup, because nothing guarantees the subscription report will come after the command response.... But in general it usually should, so we can start here to unblock people.